src/components/editor/monaco.ts

```diff
@@ -180,6 +165,26 @@ export function createMonacoEditor(
     wordWrap: opts.wordWrap,
   });
 
+  function preventStopPropagation(ev: KeyboardEvent) {
```

Comment:
I have no idea what the next 20 lines do, maybe you can add a comment explaining what it does and why it is necessary?

Reply:
Okay, I added a comment documenting that terrible hack. In one sentence, it's because monaco usually stops keyboard events from ever bubbling up the DOM, where we could catch them.